src/pages/signup/index.jsx

```diff
@@ -1,7 +1,174 @@
-import React from 'react';
+import React, { useState } from 'react';
+import bcrypt from 'bcryptjs';
+import isError from '../../utils/signUpFuncs';
 
 const SignUp = () => {
-  return <div>Hello Signup</div>;
+  const [email, setEmail] = useState('');
+  const [username, setUsername] = useState('');
+  const [phone, setPhone] = useState('');
+  const [password1, setPassword1] = useState('');
+  const [password2, setPassword2] = useState('');
+  const [errors, setErrors] = useState({});
+  const [showNoti, setShowNoti] = useState(false);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const data = {
+      email,
+      username,
+      phone,
+      password1,
+      password2,
+    };
+    const err = isError(data);
+    setErrors(err);
+
+    if (Object.keys(err).length === 0) {
+      const hash = bcrypt.hashSync(password1, 8);
+
+      const users = JSON.parse(localStorage.getItem('users'));
+      const newUser = {
+        id: users.length,
+        email,
+        username,
+        password: hash,
+        phone,
+      };
+
+      users.push(newUser);
+      localStorage.setItem('users', JSON.stringify(users));
```

Comment:
This function will usually be split into a smaller function such as "CreateAccountAPI" or something, handle submit function should only take the data, and validation and send it to the models (API) function

Reply:
Noted, I will update them accordingly